### PR TITLE
planner, sessionvar: avoid sending same task id to TiFlash

### PR DESCRIPTION
**Pull request by hanfei1991**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tics/issues/1706

For queries in same txn, the task ids sent to TiFlash might be same. I will cause some errors with a very small probability.

### What is changed and how it works?

What's Changed:
Like AllocColumnID, we add a AllocMPPTaskID function in SessionVars. If the start_ts varies, the task id will reset.

### Check List 

Tests 

- Unit test

### Release note 

-  planner, sessionvar: avoid sending same task id to TiFlash

### CODE REVIEW

**Review by qw4990 — APPROVED** [review]

**Review by fzhedu — CHANGES_REQUESTED** [review]
1. different tidb server may also conflict on task id if getting the timestamp at the same time.
2. due to killing queries by the query id, i.e. start ts, all queries in a transaction will be cancelled if just killing one of them.

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -820,6 +826,18 @@ type SessionVars struct {
 	AllowFallbackToTiKV map[kv.StoreType]struct{}
 }
 
+// AllocMPPTaskID allocates task id for mpp tasks. It will reset the task id if the query's
+// startTs is different.
+func (s *SessionVars) AllocMPPTaskID(startTS uint64) int64 {
```
Comment (fzhedu) [inline sessionctx/variable/session.go:831]: need a mutex to avoid write conflicts?

Comment (hanfei1991) [reply sessionctx/variable/session.go:831]: it's only required by one session. You can notice that every variable in SessionVars is vulnerable.

**Comment by hanfei1991** [comment]
> 1. different tidb server may also conflict on task id if getting the timestamp at the same time.
> 2. due to killing queries by the query id, i.e. start ts, all queries in a transaction will be cancelled if just killing one of them.

It's impossible that two transaction shared same timestamp.

**Review by fzhedu — APPROVED** [review]
/LGTM

**Pull request merged**